src/patterns/ProgressWizard/ProgressWizard.tsx

```diff
@@ -0,0 +1,236 @@
+ *
+ * @property startLabel, cancelLabel, backLabel, continueLabel, submitLabel - Button labels for navigation
+ *
+ * @property onSubmit - Called on final submit (receives all form data). Not compatible with the `action` prop, because it overrides native submit.
+ * @property onContinue - Called before advancing to next step (return false to block navigation)
+ * @property onBack - Called before going back (return false to block navigation)
+ * @property onCancel - Called when cancelling the wizard
+ * @property onError - Called when validation errors occur
+ *
+ * @remarks
+ *   Only one of `action` or `onSubmit` should be provided. If both are present, Typescript will get mad and `action` will be ignored.
+ */
+type ProgressWizardProps =
+  | (Omit<ProgressWizardBaseProps, 'action'> &
+      ButtonLabels &
+      Omit<CallbackProps, 'onSubmit'> & { action: string; onSubmit?: undefined })
+  | (Omit<ProgressWizardBaseProps, 'action'> &
+      ButtonLabels &
+      CallbackProps & { action?: 'Action cannot be present when `onSubmit` is supplied' });
+
+/**
+ * Main ProgressWizard component. Renders a multi-step form with validation, navigation, and custom UI.
+ *
+ * Use this component to build a step-by-step form experience. Each step is defined by a FormStep object.
+ *
+ * @param props - ProgressWizardProps
+ * @returns ReactElement with wizard UI
+ *
+ * @example
+ * <ProgressWizard
+ *   steps={[{ id: 'step1', label: 'Step 1', schema: z.object({}), componentFactory: (formContextAndOtherGoodies) => <div>Step 1</div> }]}
+ *   loadingState="idle"
+ *   onSubmit={(data) => alert(JSON.stringify(data))}
+ * />
+ */
+const ProgressWizard = forwardRef<HTMLDivElement, ProgressWizardProps>((props, ref) => {
+  const {
+    steps: propSteps,
+    defaultValues,
+    loadingState: extLoadingState,
+    customHeader,
+    hideNavigation,
+    hideProgressIndicator,
+
+    formSchema = z.object({}), // Default to empty schema to allow merging
+    action,
+
+    startLabel = 'Start',
+    cancelLabel = 'Cancel',
+    backLabel = 'Back',
+    continueLabel = 'Continue',
+    submitLabel = 'Submit',
+
+    onContinue,
+    onBack,
+    onCancel,
+    onSubmit,
+    onError,
+  } = props;
+
+  // Ensure every step has at least an empty ZodObject as its schema
+  const steps = propSteps.map((step) => ({ ...step, schema: step.schema ?? z.object({}) }));
+
+  // Nest each step's schema under its step id
+  const namespacedStepSchemas = steps.map((step) => z.object({ [step.id]: step.schema }));
+
+  // If formSchema is provided, nest it under 'formSchema'
+  const namespacedFormSchema = z.object({ formSchema });
+
+  if (steps.length > 10 && !hideProgressIndicator) {
+    console.warn(
+      'ProgressWizard',
+      'You have more than 10 steps. Consider setting `hideProgressIndicator` because it is going to look weird.',
+    );
+  }
+
+  const formId = `progress-wizard-form-${uuidv4()}`;
```

Comment:
Won't this create a new form ID on each render? Should we put this in a useRef call?
```suggestion
const formIdRef = useRef(`progress-wizard-form-${uuidv4()}`);
const formId = formIdRef.current;
```